h/static/scripts/streamer.js

```diff
@@ -155,6 +172,53 @@ function Streamer($rootScope, annotationMapper, groups, session, settings) {
     _connect();
   };
 
+  function applyPendingUpdates() {
+    var updates = Object.values(pendingUpdates).filter(function (ann) {
+      // Ignore updates to annotations that are not in the focused group
+      return ann.group === groups.focused().id;
+    });
+    var deletions = Object.keys(pendingDeletions).map(function (id) {
+      return {id: id};
+    });
+
+    annotationMapper.loadAnnotations(updates);
+    annotationMapper.unloadAnnotations(deletions);
+
+    pendingUpdates = {};
+    pendingDeletions = {};
+  }
+
+  function countPendingUpdates() {
+    return Object.keys(pendingUpdates).length;
```

Comment:
Hmm. Shall we just put a (deleted) badge on these deleted annotation cards when we disable the controls, so that the user knows why the controls have been disabled?

If we want the user to be able to remove deleted annotations without reloading the page, I'd suggest putting an (x) on them that deletes the card, but I don't think we need to support that right now.

There's a couple of problems with deletes being included in the "View n updates" button (these've been discussed on the Trello card):

First issue:

1. I am annotating and another user deletes an annotation and I receive the delete as a realtime update
2. "View n updates" button appears
3. I click it
4. The annotation, which may have been scrolled off screen, a reply in a collapsed thread, and/or in a different tab (i.e. it may not be in view), disappears. From my point of view: nothing (that I can see) happened when I clicked the button.
4. I scroll and click around trying to find the updates but there is no "updates" (no new content) to find
5. I lose trust in the view updates notification which appears to notify about spurious updates

Second issue:

I am typing a reply to an annotation, "View n updates" appears, when I click it the annotation (which was deleted by its author) disappears, presumably taking my reply with it.

We could of course try to fix this by making it only delete annotations that I don't have an unsaved reply to, but that's adding complexity, and it introduces further problems of its own ... Simpler to have one way of handling all deletes that works for both annotations that I'm in the middle of replying to and ones that I'm not.